doc/srv6/srv6-hld-v19.md

```diff
@@ -0,0 +1,552 @@
+- [1.4 Warm Reboot Requirements](#14-warm-reboot-requirements)
+- [2 Feature Design](#2-feature-design)
+- [2.1 ConfigDB Changes](#21-configdb-changes)
+- [2.2 AppDB Changes](#22-appdb-changes)
+- [2.3 Orchestration Agent Changes](#23-orchestration-agent-changes)
+- [2.4 SAI](#24-sai)
+- [2.5 CLI](#25-CLI)
+- [2.6 YANG Model](#26-yang-model )
+- [[2.7 Warm Reboot](#27-warm-reboot )
+- [3 Unit Test](#3-unit-test)
+- [4 References ](#9-references) 
+
+# Revision
+
+| Rev  |   Date   |           Author           | Change Description |
+| :--: | :------: | :------------------------: | :----------------: |
+| 0.1  | 6/5/2021 | Heidi Ou, Kumaresh Perumal |  Initial version   |
+| 0.2  |          |                            |                    |
+| 0.3  |          |                            |                    |
+
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+| ****Term**** | ****Meaning**** |
+| -------- | ----------------------------------------- |
+| BFD  | Bidirectional Forwarding Detection  |
+| BGP  | Border Gateway Protocol |
+| BSID | Binding SID |
+| G-SID  | Generalized Segment Identifier  |
+| SID  | Segment Identifier  |
+| SRH  | Segment Routing Header  |
+| SRv6 | Segment Routing IPv6  |
+| TE | Traffic Engineering |
+| uSID | Micro Segment |
+| VNI  | VXLAN Network Identifier  |
+| VRF  | Virtual Routing and Forwarding  |
+
+# About this Manual
+
+This document provides general information about the Segmentation Routing over IPv6 feature implementation in SONiC. It is based on IETF RFC 8754 and RFC 8986.
+
+# 1 Introuduction and Scope
+
+This document describes the Functionality and High level design of the SRv6 feature.
+
+SRv6 has been widely adopted as an IPv6 based SDN solution, which provides programming ability, TE capabilities,  and deployment simplicity to network administrators.  With current support from a rich ecosystem, including major ASIC manufactures,  networking vendors and open source communities, the deployment of SRv6 is accelerating. We want to add SRv6 into SONIC to benefit users in DC as well as beyond DC.  
+
+The following are some use cases for SRv6 deployment:
+
+- v4/6VPN, EVPN over best-effort
+- Traffic steering over TE policy 
+
+In SRv6 domain,  TE policy associated with SID list could be configured on headend nodes, to steer traffic with SRH encapsulation. When traffic reaches egress nodes,  the packets are processed based on local defined functions, for example SID list decapsulation and FIB lookup in a particular VRF . 
+
+# 2 Feature Requirements
+
+## 1.1 Functional Requirements
+
+This section describes the SONiC requirements for SRv6 feature in phases:
+
+At a high level the following should be supported:
+
+Phase #1
+
+​	Should be able to perform the role of SRv6 domain headend node, and endpoint node, more specific:
+
+- Support H.Encaps, H.Encaps.Red, END.B6.Encaps, END.B6.Encaps.Red
+- Support END, END.X
+- Support END.DT46
```

Comment:
Please add a short description of these endpoint behaviours in the HLD.

Reply:
done

Reply:
I don't see this addressed